src/test/integration/handlers/challenge/ShareChallengeTest.kt

```diff
@@ -84,7 +84,7 @@ class ShareChallengeTest : WordSpec() {
                     val response = handler.handleRequest(map, contxt)
                     response.statusCode shouldBe 200
 
-                    val transactionWithNewUserNamespace = JsonHelper.parse<TransactionWithNewUserNamespace>(response.body.toString())
+                    val transactionWithNewUserNamespace = JsonHelper.parse<TransactionWithNewUserNamespace>(response.body as Map<String, Any?>)
```

Comment:
could u just create a parse function that accepts 'Any' -- and have a case statement to check if its of type Map or String or whatever, and cast and call the appropriate parse? or does it not know if its a map?

Reply:
it doesn't know that its a map :/